disk/test/com/treode/disk/edit/DiskIOSpec.scala

```diff
@@ -18,62 +18,150 @@ package com.treode.disk.edit
-
-  "The PageReader" should "be able to read the string the PageWriter wrote" in {
+   "The PageReader" should "be able to read the tuple the PageWriter wrote" in {
     implicit val scheduler = StubScheduler.random()
-    val a = "this is a string"
+    val stringPickler = 
+        wrap(string,string)
+        .build(x => (x._1, x._2) )
+        .inspect( x=> (x._1, x._2) )
+
+    val a = "this is "
+    val b = "a string!"
     val readPos = 0
     val f = StubFile (1 << 20, 0)
-
-    val dw = new PageWriter (f)
+    val dsp = new PageDispatcher
+    val dw = new PageWriter (dsp, f)
     val dr = new PageReader (f)
-    val (pos, len) = dw.write (a) .expectPass()
-    val readString = dr.readString (readPos, len) .expectPass()
-    assert (a.equals (readString))
+    
+    val pDesc = PageDescriptor(0x25, stringPickler)
+    val posA = dsp.write(pDesc, 0, 0, (a,b) ) .expectPass()
+    val readString = dr.read(pDesc, posA).expectPass()
+    assert ((a,b).equals (readString))
   }
 
-  it should "write and read multiple times to disk" in {
+  "The DiskIO Reader/Writer " should " be able to write and read multiple times to disk" in {
     implicit val scheduler = StubScheduler.random()
+
+    val stringPickler = 
+        wrap(string)
+        .build(x => x )
+        .inspect( x=> x )
+
     val a = "abcdef"
     val b = "123456789"
     val startPos = 0
     val f = StubFile (1 << 20, 0)
-
-    val dw = new PageWriter (f)
+    val dsp = new PageDispatcher
+    val dw = new PageWriter (dsp,f)
     val dr = new PageReader (f)
-    val (posA, lenA) = dw.write (a) .expectPass()
-    val (posB, lenB) = dw.write (b) .expectPass()
-    val readA = dr.readString (startPos, lenA) .expectPass()
-    val readB = dr.readString (posA,     lenB) .expectPass()
+    val pDesc = PageDescriptor(0x25, stringPickler)
+    val posA = dsp.write(pDesc, 0, 0, a ) .expectPass()
+    val posB = dsp.write(pDesc, 0, 0, b ) .expectPass()
+    val readA = dr.read(pDesc, posA).expectPass()
+    val readB = dr.read(pDesc, posB).expectPass()
     assert (a.equals (readA))
     assert (b.equals (readB))
-    assert (posA == startPos + lenA)
-    assert (posB == posA + lenB)
+    assert (posA.offset == 0 )
+    assert (posB.offset == a.length + 1)
   }
 
   it should "write and read correctly out of order" in {
     implicit val scheduler = StubScheduler.random()
+
+    val stringPickler = 
+        wrap(string)
+        .build(x => x )
+        .inspect( x=> x )
+
     val a = "abcdef"
     val b = "123456789"
     val startPos = 0
     val f = StubFile (1 << 20, 0)
-
-    val dw = new PageWriter (f)
+    val dsp = new PageDispatcher
+    val dw = new PageWriter (dsp,f)
     val dr = new PageReader (f)
-    val (posA, lenA) = dw.write (a) .expectPass()
-    val (posB, lenB) = dw.write (b) .expectPass()
-    val readB = dr.readString (posA,     lenB) .expectPass()
-    val readA = dr.readString (startPos, lenA) .expectPass()
+    val pDesc = PageDescriptor(0x25, stringPickler)
```

Comment:
Just say

    PageDescriptor(0x25, Picklers.string)

Also, this is being defined multiple times. Move the definition to the head of the class, so that all the test methods can use it.